applications/KWS_Phoneme/README.md

```diff
@@ -0,0 +1,51 @@
+
+First we train a phoneme classification model which gives the phoneme transcription of the input speech snippet. For training this phoneme classifier, we use a large public speech dataset like LibriSpeech. The public dataset can be aligned (meaning get the phoneme labels for each speech snippet in the data) using Montreal Forced Aligner. We also add reverberations and additive noise to the speech samples from the public dataset to make the phoneme classifier training robust to various accents, background noise and varied environment. In this project, we predict phonemes at every 10ms which is the standard way. You can find the aligned LibriSpeech dataset we used for training here.
+
+In the second part, we use the predicted phoneme outputs from the phoneme classifier for predicting the input keyword. We train a 1 layer FastGRNN classifier to predict the keyword based on the phoneme transcription as input. Since the phoneme classifier training has been done to account for diverse accent, background noise and environments, the keyword classifier can be trained using a small number of Text-To-Speech(TTS) samples generated using any standard TTS api from cloud services like Azure, Google Cloud or AWS.
+
+This gives two advantages: (a) The phoneme model is trained to account for diverse accents and background noise settings, thus the flexible keyword classifier training requires only a small number of keyword samples, and (b) Empirically this method was able to detect keywords from as far as 9ft of distance. Further, the phoneme model has a small size of around 250k parameters and can fit on a Cortex M7 micro-controller.
+
+# Training the Phoneme Classifier
+1) Train a phoneme classification model on some public speech dataset like LibriSpeech
+2) Training speech dataset can be labelled using Montreal Force Aligner
+3) Speech snippets are convolved with reverberation files, and additive noises from YouTube or other open source are added
+4) We also add white gaussian noise of various SNRs
+
+# Training the KWS Model
+1) Our method takes as input the speech snippet and passes it through the phoneme classifier
+2) Keywords are detected by training a keyword classifier over the detected phonemes
+3) For training the keyword classifier, we use Azure and Google Text-To-Speech API to get the training data (keyword snippets)
+4) For example, if you want to train a Keyword classifier for the keywords in the Google30 dataset, generate TTS samples from the Azure/Google-Cloud/AWS API for each of the 30 keywords. The TTS samples for each keyword must be stored in a separate folder named according to the keyword. More details about how the generated TTS data should be stored are mentioned below in sample use case for classifier model training.
```

Comment:
Can we have full stops at the end of each point please? In this section as well as the above section? Also, `Keyword classifier` should be changed to `Keyword Classifier` or `keyword classifier`.